### PR TITLE
docs: reorg documentation structure for 3.0 updates

### PR DESCRIPTION
**Pull request by dcalvin**
## What have you changed? (mandatory)

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:
- Re-org the current doc structure for 3.0 updates and later migration
- Updated some outdated links
- Other minor edits

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- Improvement 

## Does this PR affect documentation (docs) or release note? (mandatory)

**YES. Please review ASAP** since it's where updates in 3.0 will be merged.

### CODE REVIEW

File: docs/how-to/deploy/using-ansible.md [hunk 1]
```diff
@@ -538,7 +538,7 @@ You can check whether the TiKV cluster has been successfully deployed using the
 curl 172.16.10.1:2379/pd/api/v1/stores
 ```
 
-If you want to try the Go client, see [Try Two Types of APIs](../clients/go-client-api.md).
+If you want to try the Go client, see [Try Two Types of APIs](../..reference/clients/go-client-api.md).
```
Comment (dcalvin) [inline docs/how-to/deploy/using-ansible.md]:
```suggestion
If you want to try the Go client, see [Try Two Types of APIs](../../reference/clients/go-client-api.md).
```

**Comment by zhangjinpeng1987** [comment]
Please sign-off.

**Comment by dcalvin** [comment]
@zhouqiang-cl Could you please look at the Jenkins Job? Thanks

**Pull request merged**